src/Compiler.php

```diff
@@ -114,18 +121,28 @@ private function getArrayProperty()
         $token = array_shift($this->request);
 
         if (!self::scanProperty($token, $array)) {
-            return false;
+            throw new Exception(
+                self::ERROR_NO_INITIAL_PROPERTY,
+                array('token' => $token),
+                self::ERROR_NO_INITIAL_PROPERTY .
+                " Error: API requests must begin with a property."
```

Comment:
This string format is the same for every exception. If we ever change it, we'll have to hunt down every string. We could define a function that takes an error code and message, and returns the formatted error string.